synapse/crypto/keyring.py

```diff
@@ -193,40 +190,42 @@ def verify_json_for_server(
             server_name,
             json_object,
             validity_time,
-            request_name,
         )
-        requests = (request,)
-        return make_deferred_yieldable(self._verify_objects(requests)[0])
+        return defer.ensureDeferred(self.process_request(request))
```

Comment:
why not make this an async function, rather than messing with `ensureDeferred`? Afaik the only reason it was defined to return a Deferred previously was that the keyring code was too scary to turn into async (all the callers, with the exception of the tests, will accept an async)